### PR TITLE
HPCC-21315 Convert FTSlave XSD to new format

### PR DESCRIPTION
**Pull request by kenrowland**
Convert FTSlave XSD to new Config Manager 2.0 format
Add ftslave component as required instance to affected components

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by kenrowland** [comment]
@xwang2713 Ming, please review. Thanks.

File: initfiles/componentfiles/configschema/xsd/ftslave_linux.xsd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                                          hpcc:presetValue="${INSTALL_DIR}/bin/ftslave" hpcc:tooltip=""/>
+                        </xs:complexType>
+                    </xs:element>
+                    <xs:element name="Notes" type="usernotes" hpcc:displayName="Notes" minOccurs="0" maxOccurs="unbounded"/>
+                </xs:sequence>
+                <xs:attributeGroup ref="buildInfo"/>
+                <xs:attribute name="name" type="xs:string" use="required" hpcc:displayName="Name"
+                              hpcc:autoGenerateType="prefix_" hpcc:autoGenerateValue="ftslave"
+                              hpcc:uniqueKey="espprocess_name" hpcc:tooltip="Name for this process"/>
+                <xs:attribute name="description" type="xs:string" hpcc:presetValue="ESP server"
```
Comment (xwang2713) [inline initfiles/componentfiles/configschema/xsd/ftslave_linux.xsd]: "ESP server" should be ftslave?

**Review by xwang2713 — COMMENTED** [review]
Assume roxie and thor processes already have the requirement for ftslave.

File: initfiles/componentfiles/configschema/xsd/ftslave_linux.xsd (hunk 1 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/ftslave_linux.xsd]: Yes. The current config manager adds one.

**Comment by kenrowland** [comment]
@richardkchapman Please merge

**Pull request merged**